### PR TITLE
meta-package-manager: fix SwiftBar plugin

### PR DESCRIPTION
**Pull request by pthariensflame**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

It broke with the latest upgrade, since it no longer knows to use the Homebrew Python.

### CODE REVIEW

File: Formula/m/meta-package-manager.rb [hunk 1]
```diff
@@ -184,6 +184,9 @@ class MetaPackageManager < Formula
   end
 
   def install
+    inreplace "meta_package_manager/bar_plugin.py",
+              "#!/usr/bin/env python3",
+              "#!#{Formula["python@3.12"].bin}/python3"
```
Comment (chenrui333) [inline Formula/m/meta-package-manager.rb:189]: can you use `rewrite_shebang detected_python_shebang, xx`

Comment (pthariensflame) [reply Formula/m/meta-package-manager.rb:189]: Oh!  I had no idea that existed!  Neat!

**Review by chenrui333 — APPROVED** [review]

**Comment by chenrui333** [comment]
Thanks @pthariensflame!

**Comment by kdeldycke** [comment]
I am thinking of extending my heuristics to locate the right Python binary as in: https://github.com/kdeldycke/meta-package-manager/blob/704d5856843991569032f32b96c767f0baa88fa2/meta_package_manager/bar_plugin.py#L102-L131

**Pull request merged**